### DISCUSSION TITLE
How do I run only the stylechecker in Maven?

### DISCUSSION THREAD

**Discussion by mengstr** [hneemann/Digital#701] [Q&A] [answered]
Apparently it's not as easy as:

```
bash-4.4# mvn checkstyle:checkstyle
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< de.neemann.digital:digital >---------------------
[INFO] Building Digital 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.0:checkstyle (default-cli) @ digital ---
[INFO] There are 25976 errors reported by Checkstyle 8.19 with sun_checks.xml ruleset.
```
since I'm a bit skeptical to truthfulness of the fact that there are 25000+ error in the code ;-)

**Comment by hneemann** [comment, marked as answer]
Hmm! I always run `mvn install`. If the style check finds something, the build process is stopped almost immediately.

The 25000+ errors are normal when using the orignal sun check rules, which are extremely strict.

**Reply by mengstr** [reply]
I use the pom you gave me 18 month ago when I had those horribly long compile times with maven in a docker container. I guess the automatic stylechecks are disabled in it and now when I called the checker manually it uses the default Sun rules instead of your own.
